diff --git a/dist/adapters/node.js b/dist/adapters/node.js
index ec81622898d583ddd8c2f4d9eaf5c0ed23776475..047ef1d45ad82e4b476a882b4ab5c8962faa0df1 100644
--- a/dist/adapters/node.js
+++ b/dist/adapters/node.js
@@ -49,7 +49,7 @@ async function createHonoServer(options) {
   }
   await mergedOptions.beforeAll?.(app);
   app.use(
-    `/${import.meta.env.REACT_ROUTER_HONO_SERVER_ASSETS_DIR}/*`,
+    `${import.meta.env.REACT_ROUTER_HONO_SERVER_BASENAME}${import.meta.env.REACT_ROUTER_HONO_SERVER_ASSETS_DIR}/*`,
     cache(60 * 60 * 24 * 365),
     // 1 year
     serveStatic({ root: clientBuildPath, ...mergedOptions.serveStaticOptions?.clientAssets })
